File: Proyecto Fin de Curso Figueroa/Frontend/src/app/services/user.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { UserResponse, UpdateUserRequest, ChangePasswordRequest } from '../models/user.model';
import { MessageResponse } from '../models/auth.model';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/usuarios`;

  getCurrentUser(): Observable<UserResponse> {
    return this.http.get<UserResponse>(`${this.apiUrl}/me`);
  }

  updateCurrentUser(request: UpdateUserRequest): Observable<UserResponse> {
    return this.http.put<UserResponse>(`${this.apiUrl}/me`, request);
  }

  changePassword(request: ChangePasswordRequest): Observable<MessageResponse> {
    return this.http.put<MessageResponse>(`${this.apiUrl}/me/password`, request);
  }

  deleteOwnAccount(): Observable<MessageResponse> {
    return this.http.delete<MessageResponse>(`${this.apiUrl}/me`);
  }

  revokeAllMySessions(): Observable<MessageResponse> {
    return this.http.delete<MessageResponse>(`${this.apiUrl}/me/sesiones`);
  }
}

File: Proyecto Fin de Curso Figueroa/Backend/src/main/java/uteq/edu/ec/artisync/service/seguridad/UserService.java
package uteq.edu.ec.artisync.service.seguridad;
import uteq.edu.ec.artisync.repository.seguridad.*;
import uteq.edu.ec.artisync.repository.perfil.*;

import uteq.edu.ec.artisync.dto.seguridad.request.ChangePasswordRequest;
import uteq.edu.ec.artisync.dto.seguridad.request.UpdateUserRequest;
import uteq.edu.ec.artisync.dto.respuesta.comun.RespuestaMensaje;
import uteq.edu.ec.artisync.dto.seguridad.response.UserResponse;

public interface UserService {
    UserResponse getCurrentUser(String correo);
    UserResponse updateCurrentUser(String correo, UpdateUserRequest request);
    RespuestaMensaje changePassword(String correo, ChangePasswordRequest request);
    RespuestaMensaje deleteOwnAccount(String correo);
    RespuestaMensaje revokeAllMySessions(String correo);
}

